### PR TITLE
Update template sysctl to fail if more than one file configure the same parameter

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Make the `sysctl` template fail if more than one file configures the same parameter
- Simplify the regex used when `sysctlval` is not used.
- Fix a bug in the remediation that prevented it from commenting parameters in `/etc/sysctl.d/*.conf` files

#### Rationale:

- RHEL8 STIG alignment

### CODE REVIEW

**Comment by ggbecker** [comment]
We probably have to test things locally for this one as container doesn't play nice with the sysctl parameters

**Comment by yuumasato** [comment]
> We probably have to test things locally for this one as container doesn't play nice with the sysctl parameters
> 
> ```
> ERROR - Terminating due to error: Return code of 'cd /root/ssgts/sysctl_net_ipv4_conf_default_accept_source_route; SHARED=/root/ssgts/shared bash -x wrong_value.fail.sh' on root@localhost is 255: Warning: Permanently added '[localhost]:45691' (ECDSA) to the list of known hosts.
> + rm -rf /usr/lib/sysctl.d/10-default-yama-scope.conf '/run/sysctl.d/*' '/etc/sysctl.d/*'
> + sed -i /net.ipv4.conf.default.accept_source_route/d /etc/sysctl.conf
> sed: can't read /etc/sysctl.conf: No such file or directory
> + echo 'net.ipv4.conf.default.accept_source_route = 1'
> + sysctl -w net.ipv4.conf.default.accept_source_route=0
> sysctl: permission denied on key "net.ipv4.conf.default.accept_source_route"
> ```

I guess the test could force existence of the the file.
The template doesn't have tests, they are all scattered on the rules, we could make this change when the tests are unified under the template.

File: shared/templates/sysctl/oval.template [hunk 1]
```diff
@@ -163,6 +146,51 @@
...
+  </ind:variable_object>
+
+  <ind:variable_state id="state_sysctl_{{{ SYSCTLID }}}_defined_in_one_file" version="1">
+    <ind:value operation="equals" datatype="int">1</ind:value>
+  </ind:variable_state>
+
+  <local_variable comment="Count unique sysctls" datatype="int" id="local_var_unique_sysctl_{{{ SYSCTLID }}}_counter" version="1">
+    <count>
+      <unique>
+        <object_component object_ref="object_static_set_sysctls_{{{ SYSCTLID }}}" item_field="filename" />
```
Comment (ggbecker) [inline shared/templates/sysctl/oval.template:166]: Two files with the same `filename` but a different `filepath` would make this check to pass.

I think we should use the `filepath` instead or combined with `filename` if `filepath` is not enough.

Comment (yuumasato) [reply shared/templates/sysctl/oval.template:166]: Good point, maybe I should remove the uniqueness of the files?
I tried with `filepath` but was running into issues, but it was before I started using `<set>`, let me try again.
I'll look into adding a test for this.

**Review by ggbecker — APPROVED** [review]
